classes/client/s3_client.php

```diff
@@ -333,4 +333,22 @@ public function define_client_section($mform, $config) {
         $mform->addHelpButton('s3_region', 'settings:aws:region', 'tool_objectfs');
         return $mform;
     }
+
+    public function copy_object_from_local_to_external_path($localpath, $contenthash) {
+
+        $externalpath = $this->get_filepath_from_hash($contenthash);
+
+        $uploader = new MultipartUploader($this->client, $localpath, [
+            'bucket' => $this->bucket,
+            'key'    => $externalpath,
+        ]);
+
+        try {
+            $result = $uploader->upload();
+            echo "Upload complete: {$result['ObjectURL']}\n";
+            return $result;
+        } catch (MultipartUploadException $e) {
+            echo $e->getMessage() . "\n";
+        }
+    }
```

Comment:
Neet to return true on success and false if anything fails to match the api of the php copy() function